File: src/main/java/org/cimsbioko/utilities/AccountUtils.java
package org.cimsbioko.utilities;

import android.accounts.Account;
import android.accounts.AccountManager;
import android.accounts.AuthenticatorException;
import android.accounts.OperationCanceledException;
import org.cimsbioko.App;
import org.cimsbioko.syncadpt.Constants;

import java.io.IOException;

public class AccountUtils {

    public static AccountManager getAccountManager() {
        return AccountManager.get(App.getApp());
    }

    private static Account[] getAccounts() {
        return getAccountManager().getAccountsByType(Constants.ACCOUNT_TYPE);
    }

    public static Account getFirstAccount() {
        Account[] accts = getAccounts();
        return accts.length > 0 ? accts[0] : null;
    }

    public static String getToken() throws AuthenticatorException, OperationCanceledException, IOException {
        return getAccountManager().blockingGetAuthToken(getFirstAccount(), Constants.AUTHTOKEN_TYPE_DEVICE, true);
    }
}

File: src/main/java/org/cimsbioko/utilities/AccountUtils.kt
package org.cimsbioko.utilities

import android.accounts.Account
import android.accounts.AccountManager
import android.accounts.AuthenticatorException
import android.accounts.OperationCanceledException
import org.cimsbioko.App
import org.cimsbioko.syncadpt.Constants
import java.io.IOException

object AccountUtils {

    private val accountManager: AccountManager
        get() = AccountManager.get(App.getApp())

    private val accounts: Array<Account>
        get() = accountManager.getAccountsByType(Constants.ACCOUNT_TYPE)

    @JvmStatic
    val firstAccount: Account?
        get() {
            return accounts.firstOrNull()
        }

    @JvmStatic
    @get:Throws(AuthenticatorException::class, OperationCanceledException::class, IOException::class)
    val token: String
        get() = accountManager.blockingGetAuthToken(firstAccount, Constants.AUTHTOKEN_TYPE_DEVICE, true)

}